api/python/provisioner/commands/check.py

```diff
@@ -621,9 +924,16 @@ def run(self, check_name: cfg.Checks = None,
         :return:
         """
         res: CheckResult = CheckResult()
+        check_lists = None
+
+        if check_name is None or check_name == cfg.GroupChecks.ALL.value:
+            check_lists = cfg.CHECKS
+
+        if not check_lists and check_name in cfg.GROUP_CHECKS:
+            check_lists = getattr(cfg, check_name.strip().upper())
 
-        if check_name is None or check_name == cfg.Checks.ALL.value:
-            for check_name in cfg.CHECKS:
+        if check_lists:
+            for check_name in check_lists:
                 _res = getattr(self,
```

Comment:
We again are changing the base logic of run method. We need to allow a user to run the check command by name of group checks as well as a single check. E.g.
```
$ provisioner check pre_checks
```
will work in current implementation but
```
$ provisioner check connectivity
```
won't work.

Reply:
With this implementation, user can run check command by name of group checks as well as a single check. For single check, check_lists is empty and hence it falls into else part. Anyway I have updated this function now for more readability.